internal/common/function/function_test.go

```diff
@@ -115,36 +122,184 @@ func TestValidateBackupSpec(t *testing.T) {
 	}
 }
 
-func TestGenerateVeleroBackupName(t *testing.T) {
-	// Max Kubernetes name length := 253
-	// hashLength := 14
-	// deduct -3: constant.VeleroBackupNamePrefix = "nab"
-	// deduct -2: two "-" in the name
-	// 253 - 14 - 3 - 2 = 234
-	const truncatedStringSize = 234
-
-	testCases := []struct {
-		namespace string
+func TestGenerateNacObjectNameWithUUID(t *testing.T) {
+	tests := []struct {
 		name      string
-		expected  string
+		namespace string
+		nabName   string
 	}{
 		{
-			namespace: "example-namespace",
-			name:      "example-name",
-			expected:  "nab-example-namespace-1cdadb729eaac1",
+			name:      "Valid names without truncation",
+			namespace: defaultStr,
+			nabName:   "my-backup",
+		},
+		{
+			name:      "Truncate nabName due to length",
+			namespace: "some",
+			nabName:   strings.Repeat("q", constant.MaximumNacObjectNameLength+10), // too long for DNS limit
+		},
+		{
+			name:      "Truncate very long namespace and very long name",
+			namespace: strings.Repeat("w", constant.MaximumNacObjectNameLength+10),
+			nabName:   strings.Repeat("e", constant.MaximumNacObjectNameLength+10),
+		},
+		{
```

Comment:
all the tests cases with empty name or namespace make no sense to me. As I commented in another place, these cases can not happen in real scenario of reconcile

Reply:
They can not, but negative testing is also correct to ensure functions are not broken by some wired change in the code. I think they don't make any harm to the testing.